cli: span used after call to Finish

Seen on `navy` during a hard shutdown:

```
panic: span used after call to Finish():
&{tracer:<nil> event:0xac2b60 Mutex:{state:0 sema:0} raw:{Context:{TraceID:7785887879600560035 SpanID:4058375174245210433 Sampled:false Baggage:map[]} ParentSpanID:0 Operation:server shutdown Start:{sec:63629515210 nsec:485499997 loc:0x2c014c0} Duration:481858792110 Tags:map[] Logs:[]} numDroppedLogs:0}

goroutine 122309 [running]:
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).maybeAssertSanityLocked(0xc4209d0140)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:31 +0x301
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).Lock(0xc4209d0140)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/debug.go:25 +0x3f
github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go.(*spanImpl).LogFields(0xc4209d0140, 0xc4218b2a00, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/opentracing/basictracer-go/span.go:123 +0xd3
github.com/cockroachdb/cockroach/pkg/util/tracing.(*TeeSpan).LogFields(0xc420186e60, 0xc4218b2a00, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/tee_tracer.go:212 +0x73
github.com/cockroachdb/cockroach/pkg/util/log.eventInternal(0x7f1978876c78, 0xc4205b9830, 0x18b0000, 0x1b0563c, 0x8, 0xc420035db0, 0x3, 0x3)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/trace.go:132 +0x299
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f1978876c78, 0xc4205b9830, 0x7f1900000001, 0x2, 0x1b0fade, 0x10, 0xc420035f28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:157 +0x460
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f1978876c78, 0xc4205b9830, 0x1, 0xc400000001, 0x1b0fade, 0x10, 0xc420035f28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:68 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Infof(0x7f1978876c78, 0xc4205b9830, 0x1b0fade, 0x10, 0xc420035f28, 0x1, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:86 +0x7e
github.com/cockroachdb/cockroach/pkg/cli.runStart.func3(0x7f1978876c78, 0xc4205b9830, 0xc4209d0000)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:478 +0x1aa
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:483 +0x920
```

--- a/pkg/cli/start.go
+++ b/pkg/cli/start.go
@@ -460,10 +460,12 @@ func runStart(cmd *cobra.Command, args []string) error {
 			serverStatusMu.Unlock()
 			if needToDrain {
 				if _, err := s.Drain(server.GracefulDrainModes); err != nil {
-					log.Warning(shutdownCtx, err)
+					// Don't use shutdownCtx because this is in a goroutine that may
+					// still be running after shutdownCtx's span has been finished.
+					log.Warning(context.Background(), err)
 				}
 			}
-			stopper.Stop(shutdownCtx)
+			stopper.Stop(context.Background())
 		}()
 	}
 
@@ -477,7 +479,7 @@ func runStart(cmd *cobra.Command, args []string) error {
 		for {
 			select {
 			case <-ticker.C:
-				log.Infof(shutdownCtx, "%d running tasks", stopper.NumTasks())
+				log.Infof(context.Background(), "%d running tasks", stopper.NumTasks())
 			case <-stopper.ShouldStop():
 				return
 			}

